src/Traits/DetectsChanges.php

```diff
@@ -91,26 +92,47 @@ public static function logChanges(Model $model): array
     {
         $changes = [];
         foreach ($model->attributesToBeLogged() as $attribute) {
-            if (str_contains($attribute, '.')) {
-                $changes += self::getRelatedModelAttributeValue($model, $attribute);
-            } else {
-                $changes += collect($model)->only($attribute)->toArray();
-            }
+            $changes += collect([$attribute => self::getModelAttributeValue($model, $attribute)])
+                ->map(function ($value) {
+                    return $value instanceof Carbon ? $value->__toString() : $value;
```

Comment:
I'm not happy with this one. But with a general `method_exists($value, '__toString')` it also converts collections and so on to json. 😕 
I'm open for improvements here!?